File: src/js/mq_consumer.js
"use strict";

var thrift = require('thrift');
var ttypes = require('./thrift-classes/first_types');
var amqp = require('amqplib');

amqp.connect("amqp://guest:guest@localhost:5672");

async function consumeMessages() {
    let conn = await amqp.connect('amqp://localhost');
    let ch = await conn.createChannel();
    let q = 'abcdefghi';
    let ok = await ch.assertQueue(q, {durable: false, autoDelete: true});

    ch.consume(q, msg => {
        let y = new ttypes.IntAndString();
        let readerTransport = new thrift.TFramedTransport(
            new Buffer(msg.content)
        );
        let readerProto = new thrift.TBinaryProtocol(readerTransport);
        y.read(readerProto);
        console.log(y);
    }, {noAck: true});
}

consumeMessages();

File: src/ruby/mq_consumer.rb
require 'bunny'
require 'faker'

$:.push('thrift-classes')

require 'first_constants'

$conn = Bunny.new("amqp://guest:guest@localhost:5672")
$conn.start
$ch = $conn.channel
$queue = $ch.queue("abcdefghi", auto_delete: true)
$deserializer = Thrift::Deserializer.new

$queue.subscribe(block: true) do |delivery_info, properties, payload|
  obj = IntAndString.new
  $deserializer.deserialize(obj, payload)
  puts "#{obj.inspect}"
end